### PR TITLE
fix pre-1.21 docker stats

### PR DESCRIPTION
**Pull request by donhcd**
This fixes a bug introduced in #15786:
- if a pre-v1.21 client requested docker stats, the daemon
  would return both an API-compatible JSON blob _and_ an API-incompatible JSON
  blob: see https://gist.github.com/donhcd/338a5b3681cd6a071629

Signed-off-by: Donald Huang don.hcd@gmail.com

### CODE REVIEW

**Comment by calavera** [comment]
can we have tests to make sure we're returning the right structs in  https://github.com/docker/docker/blob/master/integration-cli/docker_api_stats_test.go ?

**Comment by calavera** [comment]
Changes look good to me, but we need to make sure this is the right behavior.

File: daemon/stats.go [hunk 1]
```diff
@@ -106,20 +107,8 @@ func (daemon *Daemon) ContainerStats(prefixOrName string, config *ContainerStats
 						TxDropped: txDropped,
 					},
 				}
-
-				if !config.Stream && noStreamFirstFrame {
```
Comment (tiborvass) [inline daemon/stats.go]: @calavera are we sure about this?

Comment (donhcd) [inline daemon/stats.go]: this code already exists below on line 114 (in the new version)

Comment (tiborvass) [inline daemon/stats.go]: makes sense thanks

**Comment by tiborvass** [comment]
LGTM

**Comment by calavera** [comment]
LGTM

**Pull request merged**